contracts/NFT.sol

```diff
@@ -36,6 +36,17 @@ contract NFT is
         }
     }
 
+    // Overrides IERC6372 functions to make the token & governor timestamp-based
+
+    function clock() public view override returns (uint48) {
```

Comment:
I am not deep into yours ecosystem, but potentially in setup where block is on demand rather than in average fixed time (like 12s) finality is very important for a clock usage since in deep reorg it might screw the application.

Very little in network tab, but it moreover gets to the point
https://en.wikipedia.org/wiki/Clock_skew